### PR TITLE
Set all file descriptors as close on exit to prevent them from being …

### PR DESCRIPTION
**Pull request by gladish**
…inherited by WPEProcess

### CODE REVIEW

File: Source/core/DoorBell.cpp [hunk 1]
```diff
@@ -34,7 +34,11 @@ namespace Core {
     DoorBell::Connector::Connector(DoorBell& parent, const Core::NodeId& node)
         : _parent(parent)
         , _doorbell(node)
+        #ifdef __LINUX__
```
Comment (pwielders) [inline Source/core/DoorBell.cpp]: The SOCK_CLOEXEC is also available on windows so no need to use the preprocessor to diffrentiate.

File: Source/core/SocketPort.cpp [hunk 2]
```diff
@@ -644,7 +644,12 @@ namespace Core {
         }
 #endif
 
-        if ((l_Result = ::socket(localNode.Type(), SocketMode(), localNode.Extension())) == INVALID_SOCKET) {
+        #ifdef __LINUX__
```
Comment (pwielders) [inline Source/core/SocketPort.cpp]: Windows does support SOCK_CLOEXEC as well so lets not diffrentiate her

File: Source/core/SocketPort.cpp [hunk 3]
```diff
@@ -1132,7 +1138,11 @@ namespace Core {
         socklen_t size = sizeof(address);
         SOCKET result;
 
+        #ifdef __LINUX__
+        if ((result = ::accept4(m_Socket, (struct sockaddr*)&address, &size, SOCK_CLOEXEC)) != SOCKET_ERROR) {
+        #else
```
Comment (pwielders) [inline Source/core/SocketPort.cpp]: Could not find the accept4 in the windows documentation that quickly, but suggest we remove the preprocessor stuff anyway. I will test it on windows and if I run into an issue I will just create a local accept4 method in this compilation unit that will push the SOCK_CLOEXEC option onto the socket. I am afraid if I am not running into a compilation issue, we forget about this change.

**Review by pwielders — CHANGES_REQUESTED** [review]
Looks good, just would like to drop the __LINUX_ guards you added. If it hurst in windows (and I think only the accept4 will) we will fix it so that also on windows, connecting sockets get the SOCK_CLOEXEC flag.

**Review by pwielders — APPROVED** [review]
Looks good to me. Waiting for @HaseenaSainul to make sure that we do not introduce any regression if she reports no regression good to merge!

**Review by HaseenaSainul — APPROVED** [review]
LGTM

**Pull request merged**